domain-models-runtime/src/test/java/org/folio/rest/tools/utils/TenantLoadingTest.java

```diff
@@ -560,6 +564,33 @@ public void test404IdRaw(TestContext context) {
     });
   }
 
+  @Test
+  public void testGetIdBase(TestContext context) {
+    Promise promise = Promise.promise();
+    context.assertEquals("/a", TenantLoading.getIdBase("/path/a.json", promise.future()));
+    context.assertFalse(promise.future().isComplete());
+    promise = Promise.promise();
+    context.assertEquals(null, TenantLoading.getIdBase("a.json", promise.future()));
+    context.assertTrue(promise.future().isComplete());
+    context.assertEquals("No basename for a.json", promise.future().cause().getMessage());
+    promise = Promise.promise();
+    context.assertEquals(null, TenantLoading.getIdBase("a", promise.future()));
+    context.assertTrue(promise.future().isComplete());
+    promise = Promise.promise();
+    context.assertEquals("/a", TenantLoading.getIdBase("/path/a", promise.future()));
+    context.assertFalse(promise.future().isComplete());
+    promise = Promise.promise();
+    context.assertEquals("/c", TenantLoading.getIdBase("/a.b/c", promise.future()));
+    context.assertFalse(promise.future().isComplete());
+  }
```

Comment:
`testGetIdBase` contains unnecessary boilerplate code that makes it difficult to understand what is tested. How about something like this?

    private void assertGetIdBase(String path, String expectedBase) {
      Future<Void> future = Promise.<Void>promise().future();
      assertEquals(expectedBase, TenantLoading.getIdBase(path, future));
      assertFalse(future.isComplete());
    }

    private void assertGetIdBaseFail(String path) {
      Future<Void> future = Promise.<Void>promise().future();
      assertEquals(null, TenantLoading.getIdBase(path, future));
      assertTrue(future.failed());
      assertEquals("No basename for " + path, future.cause().getMessage());
    }

    @Test
    public void testGetIdBase() {
      assertGetIdBase("/path/a.json", "/a");
      assertGetIdBase("/path/a", "/a");
      assertGetIdBase("/a.b/c", "/c");
      assertGetIdBaseFail("a.json");
      assertGetIdBaseFail("a");
    }